data-gathering/bot.py

```diff
@@ -1,726 +1,529 @@
+   f = open(file_to_be_analysed, 'r', encoding= utf_required )  #change this to file to read as an option from users
 
-
-
-
-# analyze articles
-def analysis():
-
-   # Open the "query.csv" file in read mode
-   f = open('query.csv', "r", encoding='utf-8')
-
-# Read all lines from the file into a list
-   vox = f.readlines()   
+   articles = f.readlines()   
     
    # delete the contents of the file before starting
-   
-   results = open('resultati.txt',"w")
-   
-   # Truncate the "results.txt" file to remove existing content
+   results = open('results.txt',"w")
    results.truncate(0)
-   
    results.close()
 
-# Iterate through each article in the list of articles
-   for article in vox:
-      
-      # Open the "results.txt" file in append mode
-      results = open('resultati.txt', 'a')  # open the file in append mode
+   for article in articles:
+      results = open('results.txt', 'a')  # open the file in append mode
 
       flag = 1
 
-      # Remove the newline character at the end of the article
       article = article[:-1]
-
-      # Replace spaces in the article title with underscores
-      article = article.replace(" ","_")
-
-      ris = ""
-
+      article= article.replace(" ","_") # Wikipedia page titles are case-sensitive and spaces in page titles should be replaced with underscores.
+      result = ""
       wikitext = ""
 
-
-      # Replace spaces in the article title with underscores and quote the article
-      article2 = urllib.parse.quote(article)
-      article = article.replace(" ","_")
+      article2 = urllib.parse.quote(article) 
 
 
       try:
 
-         # Construct the Wikipedia API URL for parsing wikitext
-        url = "https://"+language+".wikipedia.org/w/api.php?action=parse&page=" + article2 + "&prop=wikitext&formatversion=2&format=json"
-      
-      # Open the URL and load JSON data
-        json_url = urlopen(url)
-
-        data = json.loads(json_url.read())
-      
-      # Extract wikitext from the JSON response
-        wikitext = data["parse"]["wikitext"]
+         url = "https://"+language+".wikipedia.org/w/api.php?action=parse&page=" + article2 + "&prop=wikitext&formatversion=2&format=json"
+         json_url = urlopen(url)
 
-      # If the article is a redirect, update the article title
-        if "#REDIRECT"  in wikitext:
+         data = json.loads(json_url.read())
 
-     #   print (wikitext)
+         # for debuging
+         # if "error" in data:
+         #    error_message = data["error"]["info"]
+         #    print(f"Error for article '{article}': {error_message}")
+         #    result = result + article + "\t" + error_message  # Include the error message in the result
+         # else:
+         #    wikitext = data["parse"]["wikitext"]
 
-          article2 = wikitext[wikitext.find("[[")+2:]
+         wikitext = data["parse"]["wikitext"]
 
-          article2 = article2[:article2.find("]]")]
+         if "#RINVIA"  in wikitext:
```

Comment:
Is #RINVIA an equivalent to #REDIRECT? Does it work for all languages?

Reply:
No, it doesnt work for all languages. I'll add checking for "REDIRECT" too